render-ws-spark-client/src/main/java/org/janelia/render/client/spark/pipeline/AlignmentPipelineParameters.java

```diff
@@ -63,62 +73,48 @@ public MultiProjectParameters getMultiProject() {
         return multiProject;
     }
 
-    public boolean hasMipmapParameters() {
-        return mipmap != null;
-    }
-
     public MipmapParameters getMipmap() {
         return mipmap;
     }
 
-    public boolean hasMatchParameters() {
-        return (matchRunList != null) && (! matchRunList.isEmpty());
-    }
-
     public List<MatchRunParameters> getMatchRunList() {
         return matchRunList;
     }
 
-    public boolean hasMfovMontagePatchParameters() {
-        return (mfovMontagePatch != null);
-    }
-
     public MFOVMontageMatchPatchParameters getMfovMontagePatch() {
         return mfovMontagePatch;
     }
 
-    public boolean hasUnconnectedCrossMfovParameters() {
-        return (unconnectedCrossMfov != null);
-    }
-
     public UnconnectedCrossMFOVParameters getUnconnectedCrossMfov() {
         return unconnectedCrossMfov;
     }
 
-    public boolean hasTileClusterParameters() {
-        return (tileCluster != null);
-    }
-
     public TileClusterParameters getTileCluster() {
         return tileCluster;
     }
 
-    public boolean hasMatchCopyParameters() {
-        return (matchCopy != null);
-    }
-
     public MatchCopyParameters getMatchCopy() {
         return matchCopy;
     }
 
-    public boolean hasAffineBlockSolverSetup() {
-        return (affineBlockSolverSetup != null);
-    }
-
     public AffineBlockSolverSetup getAffineBlockSolverSetup() {
         return affineBlockSolverSetup;
     }
 
+    public List<AlignmentPipelineStep> buildStepClients()
+            throws IllegalArgumentException {
+
+        if ((pipelineSteps == null) || pipelineSteps.isEmpty()) {
+            throw new IllegalArgumentException("no pipeline steps defined");
+        }
```

Comment:
Just to make things clear: we go for "fail fast" rather than a smooth execution (in which case we could return an empty list), right? In this case, the check for missing parameters should probably be done right after JSON deserialization.
The exception should also be noted in the doc-string for this method.

Reply:
Yes - fail fast.  This check is effectively done right after deserialization.  Added javadoc - good point.